src/transaction/boot_sr.c

```diff
@@ -2263,6 +2263,18 @@ boot_restart_server (THREAD_ENTRY * thread_p, bool print_restart, const char *db
       goto error;
     }
 
+#if defined(SERVER_MODE)
+  if (!HA_DISABLED ())
+    {
+      // This must be called after init_server_type () because cub_master have to know which type of server it is.
+      error_code = css_register_ha_server (db_name);
```

Comment:
Need a comment shortly about why it's needed here. At least the mention that it has to be before recovery.